ml-algorithms/build.gradle

```diff
@@ -42,22 +42,22 @@ dependencies {
     testImplementation group: 'org.mockito', name: 'mockito-core', version: '5.7.0'
     implementation group: 'com.google.guava', name: 'guava', version: '32.1.2-jre'
     implementation group: 'com.google.code.gson', name: 'gson', version: '2.10.1'
-    implementation platform("ai.djl:bom:0.21.0")
-    implementation group: 'ai.djl.pytorch', name: 'pytorch-model-zoo', version: '0.21.0'
+    implementation platform("ai.djl:bom:0.28.0")
+    implementation group: 'ai.djl.pytorch', name: 'pytorch-model-zoo'
     implementation group: 'ai.djl', name: 'api'
     implementation group: 'ai.djl.huggingface', name: 'tokenizers'
-    implementation("ai.djl.onnxruntime:onnxruntime-engine:0.21.0") {
+    implementation("ai.djl.onnxruntime:onnxruntime-engine") {
         exclude group: "com.microsoft.onnxruntime", module: "onnxruntime"
     }
     def os = DefaultNativePlatform.currentOperatingSystem
     //arm/macos doesn't support GPU
     if (os.macOsX || System.getProperty("os.arch") == "aarch64") {
         dependencies {
-            implementation "com.microsoft.onnxruntime:onnxruntime:1.14.0"
+            implementation "com.microsoft.onnxruntime:onnxruntime:1.16.3!!"
```

Comment:
What's ! means here?

Reply:
The !! syntax is used to enforce the exact version of a dependency in Gradle, ensuring that no other version is used in the build process, which can help avoid compatibility issues and ensure consistency across different builds.

Reply:
@b4sjoo https://docs.gradle.org/current/userguide/single_versions.html#ex-shorthand-notation-for-strict-dependencies